In [47]:
import pandas as pd

In [48]:
df_mrd=pd.read_csv('labeledTrainData.tsv',sep='\t')

In [49]:
df_mrd2=df_mrd.groupby(['sentiment'])['sentiment'].count()
print(df_mrd2)

sentiment
0    12500
1    12500
Name: sentiment, dtype: int64


In [50]:
from textblob import TextBlob
df_mrd['sentiment_pol'] = df_mrd['review'].apply(lambda tweet: TextBlob(tweet).polarity)
df_mrd.head()

id  sentiment                                             review  \
0  5814_8          1  With all this stuff going down at the moment w...   
1  2381_9          1  \The Classic War of the Worlds\" by Timothy Hi...   
2  7759_3          0  The film starts with a manager (Nicholas Bell)...   
3  3630_4          0  It must be assumed that those who praised this...   
4  9495_8          1  Superbly trashy and wondrously unpretentious 8...   

   sentiment_pol  
0       0.001277  
1       0.256349  
2      -0.053941  
3       0.134753  
4      -0.024842

In [51]:
df_mrd['sentiment_calc'] = df_mrd['sentiment_pol'].apply(lambda x: 1 if x >= 0 else 0)
df_mrd.head()

id  sentiment                                             review  \
0  5814_8          1  With all this stuff going down at the moment w...   
1  2381_9          1  \The Classic War of the Worlds\" by Timothy Hi...   
2  7759_3          0  The film starts with a manager (Nicholas Bell)...   
3  3630_4          0  It must be assumed that those who praised this...   
4  9495_8          1  Superbly trashy and wondrously unpretentious 8...   

   sentiment_pol  sentiment_calc  
0       0.001277               1  
1       0.256349               1  
2      -0.053941               0  
3       0.134753               1  
4      -0.024842               0

In [52]:
from sklearn.metrics import accuracy_score
accuracy_score(df_mrd['sentiment'],df_mrd['sentiment_calc'])

0.68524

Here the accuracy of our model is 68.5 % which is better than a random guess which has aprobability of 0.5 or 50%

In [53]:
import nltk
from nltk.sentiment.vader import SentimentIntensityAnalyzer
sia = SentimentIntensityAnalyzer()
df_mrd['vader_pred'] = df_mrd['review'].apply(lambda x: 1 if sia.polarity_scores(x)['compound'] >= 0 else 0)


In [54]:
from sklearn.metrics import accuracy_score
accuracy_score(df_mrd['sentiment'],df_mrd['vader_pred'])

0.69356

Here the accuracy of our model is 69.3 % which is better than a random and slightly better than TextBlob